src/pool/pool_disjoint.c

```diff
@@ -233,7 +231,11 @@ static umf_result_t pool_unregister_slab(disjoint_pool_t *pool, slab_t *slab) {
     // TODO ASSERT_IS_ALIGNED((uintptr_t)slab_addr, bucket->size);
     LOG_DEBUG("slab: %p, start: %p", (void *)slab, slab_addr);
 
-    critnib_remove(slabs, (uintptr_t)slab_addr);
+    void *ref_slabs = NULL;
+    critnib_remove(slabs, (uintptr_t)slab_addr, &ref_slabs);
```

Comment:
What do you think about an API change where, if I pass NULL as the last argument, the `critnib_release` would be called automatically at the end of the `critnib_remove` call? This would mean the change here would simply be `critnib_remove(slabs, (uintptr_t)slab_addr, NULL);`

Reply:
It would be not consistent and misleading. We can add `critnib_remove_release(slabs, (uintptr_t)slab_addr);` instead.

Reply:
I'm ok with critnib_remove_release

Reply:
`critnib_remove_release()` added.
Done